contracts/earn/strategies/SpookySwap/StrategyLPV2.sol

```diff
@@ -0,0 +1,211 @@
+
+import '../../../interfaces/SpookySwap/IMasterChefV2.sol';
+import '../../../interfaces/SpookySwap/ISpookyPair.sol';
+import '../../../interfaces/PancakeSwap/IPancakeRouter.sol';
+import '../../../interfaces/IWooAccessManager.sol';
+import '../../../interfaces/IStrategy.sol';
+
+import '../BaseStrategy.sol';
+
+contract StrategySpookySwapLPV2 is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ----- State Variables ----- */
+
+    uint256 public immutable pid;
+
+    address[] public rewardToLP0Route;
+    address[] public rewardToLP1Route;
+
+    address public lpToken0;
+    address public lpToken1;
+
+    address public constant reward = address(0x841FAD6EAe12c286d1Fd18d1d525DFfA75C7EFFE); // BOO
+    address public constant uniRouter = address(0xF491e7B69E4244ad4002BC14e878a34207E38c29);
+    address public constant masterChefV2 = address(0x18b4f774fdC7BF685daeeF66c2990b1dDd9ea6aD);
+
+    constructor(
+        address _vault,
+        address _accessManager,
+        uint256 _pid,
+        address[] memory _rewardToLP0Route,
+        address[] memory _rewardToLP1Route
+    ) public BaseStrategy(_vault, _accessManager) {
+        pid = _pid;
+        rewardToLP0Route = _rewardToLP0Route;
+        rewardToLP1Route = _rewardToLP1Route;
+
+        if (_rewardToLP0Route.length == 0) {
+            lpToken0 = reward;
+        } else {
+            require(_rewardToLP0Route[0] == reward);
+            lpToken0 = _rewardToLP0Route[_rewardToLP0Route.length - 1];
+        }
+        if (_rewardToLP1Route.length == 0) {
+            lpToken1 = reward;
+        } else {
+            require(_rewardToLP1Route[0] == reward);
+            lpToken1 = _rewardToLP1Route[_rewardToLP1Route.length - 1];
+        }
+
+        require(
+            ISpookyPair(want).token0() == lpToken0 || ISpookyPair(want).token0() == lpToken1,
+            'StrategySpookySwapLPV2: token0_INVALID'
+        );
+        require(
+            ISpookyPair(want).token1() == lpToken0 || ISpookyPair(want).token1() == lpToken1,
+            'StrategySpookySwapLPV2: token1_INVALID'
+        );
+
+        address lpToken = IMasterChefV2(masterChefV2).lpToken(_pid);
+        require(lpToken == want, 'StrategySpookySwapLPV2: _pid_INVALID');
+
+        _giveAllowances();
+    }
+
+    /* ----- External Functions ----- */
+
+    function withdraw(uint256 amount) external override nonReentrant {
+        require(msg.sender == vault, 'StrategySpookySwapLPV2: NOT_VAULT');
+
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance < amount) {
+            IMasterChefV2(masterChefV2).withdraw(pid, amount.sub(wantBalance));
+            wantBalance = IERC20(want).balanceOf(address(this));
+        }
+
+        // just in case the decimal precision for the very left staking amount
+        uint256 withdrawAmount = amount < wantBalance ? amount : wantBalance;
```

Comment:
require(wantBalance >= amount.mul(9999).div(10000), 'StrategySJoe: WITHDRAW_INSUFF_AMOUNT');
        uint256 withdrawAmount = amount < wantBalance ? amount : wantBalance;

I'd suggest to add this require . (Okay to go with 999/1000